sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/ConsumerSpEL.java

```diff
@@ -57,4 +74,31 @@ public void evaluateAssign(Consumer consumer, Collection<TopicPartition> topicPa
     mapContext.setVariable("tp", topicPartitions);
     assignExpression.getValue(mapContext);
   }
+
+  private boolean hasTimestamp(){
+    boolean hasEventTimestamp = false;
+    try {
+      hasEventTimestamp = timestampMethod != null
+          && timestampMethod.getReturnType().equals(Long.TYPE);
+    } catch (SecurityException e) {
+      hasEventTimestamp = false;
+    }
+    return hasEventTimestamp;
+  }
+
+  public long getEventTimestamp(ConsumerRecord<byte[], byte[]> rawRecord) {
```

Comment:
Rename ito getTimestamp or getRecordTimestamp()? Rest of KafkaIO refers each messages as 'record' rather than an 'event.